docs/src/solutions.md

```diff
@@ -0,0 +1,100 @@
+Querying Solutions
+==================
+
+So far we have seen all the elements and constructs related to writing a JuMP
+optimization model. In this section we reach the point of what to do
+with a solved problem. Suppose your model is named `model`, right after the
+call to `JuMP.optimize!(model)` (which might take a while) it's possible to
+ask JuMP questions about the finished optimization step. Typical questions
+include: Why has the optimization process stopped? Do I have a solution to my
+problem? Is it optimal? Do I have a dual solution?
+
+JuMP follows closely the concepts defined in [MathOptInterface (MOI)](https://github.com/JuliaOpt/MathOptInterface.jl)
+to answer user questions about a finished call to `JuMP.optimize!(model)`.
+There are 4 main steps in querying a solution:
+
+* Firstly, we obtain the [`termination_status`](@ref) which will tell the user the reason why
+optimization stopped. An optimal solution could have been found, or the problem
+was proven to be infeasible, or the solver could not converge or JuMP got some
+solver specific error etc.
+* Secondly, we should figure out the [`primal_status`](@ref) and [`dual_status`](@ref), which will tell us what
+kind of result do we have for our primal and dual solution. Maybe we have an
+optimal primal-dual pair, or maybe just primal variable, or nothing at all,
+or some infesability cetificate or even more.
+* The third step is more straight forward, we should figure out if there really
+is something available to be queryed in the primal solution with [`has_values`](@ref)
+and in the dual soltion with [`has_duals`](@ref).
+* Finally, we can [`JuMP.value`](@ref) and [`JuMP.dual`](@ref) to obtain primal and dual values of
+the optimization variables and constraints.
+
+## Termination Statuses
+
+The possible reason why the optimization of `model` was finished is given by the call
+
+```julia
+JuMP.termination_status(model)
+```
+
+This function will return an instance of the `MOI.TerminationStatus` `enum`. All the possible outcome can be seen by calling help:
+
+```julia
+?MOI.TerminationStatus
+```
+
+```@docs
+JuMP.termination_status
+MOI.TerminationStatus
+```
+
+TODO
+Do we add an exautive table here?
```

Comment:
We can maybe just add a link to http://www.juliaopt.org/MathOptInterface.jl/stable/apimanual/#Common-status-situations-1